src/main/scala/org/broadinstitute/dsde/workbench/avram/db/DbReference.scala

```diff
@@ -0,0 +1,45 @@
+package org.broadinstitute.dsde.workbench.avram.db
+
+import org.apache.commons.dbcp2.BasicDataSource
+import org.broadinstitute.dsde.workbench.avram.config.DbcpDataSourceConfig
+import slick.dbio.DBIO
+import slick.jdbc.{JdbcProfile, TransactionIsolation}
+import scala.concurrent.{ExecutionContext, Future}
+
+import AvramPostgresProfile.api._
+
+case class DbReference(private val dataSourceConfig: DbcpDataSourceConfig)(implicit val executionContext: ExecutionContext) {
+
+  val dataAccess = new DataAccess(AvramPostgresProfile)
+
+  val ds = new BasicDataSource()
+  ds.setDriverClassName(dataSourceConfig.driverClassName)
+  ds.setUrl(dataSourceConfig.url)
+  ds.setUsername(dataSourceConfig.username)
+  ds.setPassword(dataSourceConfig.password)
+  ds.setMaxTotal(dataSourceConfig.maxTotal)
+
+  val database = Database.forDataSource(ds, Option(ds.getMaxTotal), AsyncExecutor("Avram Executor", dataSourceConfig.slickNumThreads, dataSourceConfig.slickQueueSize))
+
+  def inTransaction[T](f: (DataAccess) => DBIO[T], isolationLevel: TransactionIsolation = TransactionIsolation.RepeatableRead): Future[T] = {
+    import dataAccess.profile.api._
+    database.run(f(dataAccess).transactionally.withTransactionIsolation(isolationLevel))
+  }
+}
+
+class DataAccess(val profile: JdbcProfile)(implicit val executionContext: ExecutionContext) extends AllComponents {
+
+  def truncateAll(): DBIO[Int] = {
```

Comment:
This should only be called from tests. It would be great to find a way to have this function only live in src/test.